reports/GitalyFail/GitalyFail.tla

```diff
@@ -1,137 +1,117 @@
+    rmGenerations, targetGeneration
+
+\* Invariants
+TypeOK == \A r \in RM: rmState[r] \in [RM -> {"init", "working", "prepared", "committed", "aborted"}]
+
+Consistent == \A r \in RM : ~( \/ (/\ rmState[r] = "aborted"
+                                   /\ rmGenerations[r] = 1)
+                               \/ (/\ rmState[r] = "committed"
+                                   /\ rmGenerations[r] = 0)
+                              )
+
+-------------------------------------------------------------------------------
+========
+
+MasterNode = Head(RM)
+            
+GFInit ==   /\ rmState = [r \in RM |-> "init"]            
+            /\ rmGenerations = [r \in RM |-> 0]
+            /\ targetGeneration = 0
+            /\ rmVotedYes = {}
+            /\ rmVotedNo = {}
+            /\ rmAck = {}
+            /\ rmPrepared = {}
+            /\ rmCommitted = {}
+            /\ rmAborted = {}
+
+\* Step 1.1 Preafect call RM and swith it to working state
+RMRcvPack(r) == /\ rmState[r] = "init"
+                /\ rmState' = [rmState EXCEPT ![r] = "working"] \* All the Nodes are answered
+                
+\* Step 2.1 Node calls Preafect and send Vote
+NodeVote(r) == /\ r \notin rmVotedYes
+               /\ r \notin rmVotedNo
+               /\ rmState[r] = "working"
+               /\ (\/ (/\ rmState' = [rmState EXCEPT ![r] = "prepared"]
+                       /\ rmVotedYes' = rmVotedYes \union r
+                       /\ rmPrepared' = rmPrepared \union r) \* Prepared
+                   \/ (/\ rmState' = [rmState EXCEPT ![r] = "aborted"]
+                       /\ rmVotedNo' = rmVotedNo \union r
+                       /\ rmAborted' = rmAborted \union r)  \* Aborted
+                   )            
+
+\* Step 2.2 Preafect await for Quorum and send its decision
+isQuorumYes == /\ MasterNode \in rmVotedYes
+               /\ rmState[MasterNode] = "prepared" \* Doublecheck
+               /\ Len(rmVotedYes) * 2 > Len(RM)
+
+isQuorumNo  == \/ (/\ MasterNode \in rmVotedNo
+                   /\ rmState[MasterNode] = "aborted" \* Doublecheck
+               \/ Len(rmVotedNo) * 2 > Len(RM)              
+
+RMRcvCommit(r) == /\ isQuorumYes
+                  /\ rmState' = [rmState EXCEPT ![r] = "committed"]
+                  \* Send Ack or not send
+                  /\ (\/ rmAck' = rmAck \union r \* Answered to Preafect and counted by it
+                      \/ UNCHANGED << rmAck >>) \* Preafect didn't receive acknowlege
+
+RMRcvAbort(r) == /\ isQuorumNo
+                 /\ rmState' = [rmState EXCEPT ![r] = "aborted"]
+                 /\ (\/ rmAck' = rmAck \union r \* Answered to Preafect and counted by it
+                      \/ UNCHANGED << rmAck >>) \* Preafect didn't receive acknowlege
+
+\* Step 3.1 Preafect await all acks and calls DB and increase generations.
+
+\* A trick to wait for all nodes are done their work
+isTimeOut == \A r \in RM:
+            \/ rmState[r] = "committed"
+            \/ rmState[r] = "aborted"

```

Comment:
RM node doesn't know the state of  each other, the valid condition for time out would be that node is in `prepared` state